server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/BlobManagerImpl.java

```diff
@@ -62,11 +71,29 @@ public BlobId toBlobId(MessageId messageId) {
 
     @Override
     public Blob retrieve(BlobId blobId, MailboxSession mailboxSession) throws MailboxException, BlobNotFoundException {
-        return getBlobFromAttachment(blobId, mailboxSession)
-                .orElseGet(() -> getBlobFromMessage(blobId, mailboxSession)
+        return getBlobFromUpload(blobId, mailboxSession)
```

Comment:
```java
getBlobFromUpload(blobId, mailboxSession)
.or(Throwing.supplier(() -> getBlobFromAttachment(blobId, mailboxSession)).sneakyThrow())
.or(() -> getBlobFromMessage(blobId, mailboxSession))
.orElseThrow(() -> new BlobNotFoundException(blobId));
```
Not necessary, but it is easier to understand

Reply:
`Throwing.supplier(() -> getBlobFromAttachment(blobId, mailboxSession)).sneakyThrow().get()`
Need `.get()` for subscribe

Reply:
https://github.com/apache/james-project/pull/584 reactified this, making this awesome!

https://github.com/apache/james-project/pull/584